sigstore-java/src/main/java/dev/sigstore/tuf/Updater.java

```diff
@@ -221,13 +225,28 @@ void verifyDelegate(
           } else {
             pubKey = Keys.constructTufPublicKey(Hex.decode(publicKeyContents), key.getScheme());
           }
-          byte[] signatureBytes = Hex.decode(signature.getSignature());
           try {
+            // while we error on keys that are not readable, we are intentionally more permissive
+            // about signatures. If for ANY reason (except unparsed keys) we cannot validate a
+            // signature, we continue as long as we find enough valid signatures within the
+            // threshold
+            byte[] signatureBytes = Hex.decode(signature.getSignature());
             if (verifiers.newVerifier(pubKey).verify(verificationMaterial, signatureBytes)) {
               goodSigs.add(signature.getKeyId());
             }
-          } catch (SignatureException e) {
-            throw new TufException(e);
+          } catch (SignatureException
+              | DecoderException
+              | NoSuchAlgorithmException
+              | InvalidKeyException e) {
+            log.log(
+                Level.FINE,
```

Comment:
'warn' maybe? seems like a data problem that you'd want to know about.

Reply:
I think these errors can occur naturally with no ill effect. Maybe we should separate expected from unexpected here? And log accordingly?

@jku presumably just the `""` case is expected?

Reply:
Without knowing details of the exceptions:
* In a generic TUF client a signature mismatch should not generate warnings visible to users by default
* since this is specifically a sigstore TUF client, you could use the knowledge that only empty strings are used as "signatures": I think I would not warn even in other signature mismatch cases though
* the other errors sound like issues that maybe should result in something visible to user

Reply:
alright moved things around.